_documentation/05-User-Manual.md

```diff
@@ -201,3 +201,125 @@ Apart from configuring printing defaults for the printer, these are the other pr
 3. **Print Test Page**
 
     This option helps in testing printer functionality, by issuing a print command for a sample test page. 
+
+## Working with Command-Line
+
+Before start working with your device, you must know about the capabilities of your printer/scanner by viewing the set of sub-commands and options supported by your printer/scanner. For retrieving this information, you need to pass the `--help` argument:
+
+    <application-name> --help
+
+Although, each application may have a different usage-callback function, the default list of sub-commands and options available with all applications is shown below.
+
+    Usage: <application-name> SUB-COMMAND [OPTIONS] [FILENAME]
+        <application-name> [OPTIONS] [FILENAME]
+        <application-name> [OPTIONS] -
+
+    Sub-commands:
+    *add PRINTER      Add a printer.
+    *cancel           Cancel one or more jobs.
+    *default          Set the default printer.
+    *delete           Delete a printer.
+    *devices          List devices.
+    *drivers          List drivers.
+    *jobs             List jobs.
+    *modify           Modify a printer.
+    *options          List printer options.
+    *printers         List printers.
+    *server           Run a server.
+    *shutdown         Shutdown a running server.
+    *status           Show server/printer/job status.
+    submit           Submit a file for printing.
+
+    Options:
+    -a               Cancel all jobs (cancel).
+    -d PRINTER       Specify printer.
+    -j JOB-ID        Specify job ID (cancel).
+    -m DRIVER-NAME   Specify driver (add/modify).
+    -n COPIES        Specify number of copies (submit).
+    -o NAME=VALUE    Specify option (add,modify,server,submit).
+    -u URI           Specify ipp: or ipps: printer/server.
+    -v DEVICE-URI    Specify socket: or usb: device (add/modify).
+
+### Server Daemon
+
+* **Start Server**
+    <br>Before adding devices and submitting jobs, you must start the application as a server daemon:
+
+        <application-name> server [-o <NAME> = <VALUE>]
+
+* **Close Server**
+    <br>You can close the daemon using the following command:
+
+        <application-name> shutdown
+
+### Devices
+
+* **Adding Device**
+    <br> Add a printer queue.
+
+        <application-name> add -d <DEVICE_NAME> -m <DRIVER-NAME> -v <DEVICE-URI> [-o <NAME> = <VALUE>]
+
```

Comment:
Tell the user here that the he can obtain the possible '<DRIVER NAME>' values via the call with the "drivers" subcommand and the possible '<DEVICE-URI>' values via the call with "devices" subcommand.

Reply:
Added, thanks for the feedback :-)